### DISCUSSION TITLE
Bug looking up UUID with h2 on 2.9.0

### DISCUSSION THREAD

**Discussion by pptheberge** [quarkusio/quarkus#25535] [Community]
It seems like the h2 upgrade in 2.9.0 broke looking up UUID columns when they're automatically created.

Prior to h2 version 2, BINARY was just another name for VARBINARY., after version 2,  BINARY and VARBINARY are now separate types with the former populating the database with 0 padded values. Since the UUID java type gets represented by default as a BINARY, all lookups fail. 

```
public class SomeObject {
	@Id
	UUID id;
	//….
}

// stringId passed from API call
 UUID id = UUID.fromString(stringId);
SomeObject object = repository.findById(id);
 // object is always null with 2.9.0

```
Shouldn’t the default behavior be to create h2 columns using the UUID database type?

By making this simple change, everything works.
```
public class SomeObject {
	@Id
	@Column(columnDefinition = "uuid")
	UUID id;
	//….
}

```

Thanks,

**Comment by WilliamWishart** [comment]
What import is required to support @ColumnDefinition in this case?

**Comment by pptheberge** [comment]
Sorry, that was a typo when I wrote it. It should say

`    @Column(columnDefinition = "uuid")`

**Reply by gsmet** [reply]
FWIW this will be the default mapping of Hibernate ORM 6 (it's not possible to change the default mapping easily in ORM 5) so when Quarkus moves to Hibernate ORM 6 (but it's still several months away), it will be automatic.

**Comment by gsmet** [comment]
I had the issue when upgrading and I fixed it this way: https://github.com/quarkusio/quarkus/pull/24663/files#diff-e94ad5b9df019618c0f1b4db509b28dcd8ebfb74745f95cb5f06eac93fc2cbf7 .

I agree it's not exactly ideal if you have a lot of UUIDs in your code base. @beikov it seems to be a recurrent problem, I don't know if we can improve the situation?

**Reply by beikov** [reply]
FTR, see the discussion on [Zulip](https://hibernate.zulipchat.com/#narrow/stream/132094-hibernate-orm-dev/topic/H2.202.20and.20UUID) about this. TLDR, Quarkus can provide a custom dialect to improve the situation, but in ORM 5.x we can't really do anything. The issue is fixed in 6.0 though.
